apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1495,14 +1495,17 @@ func (u UniterAPI) readOneUnitSettings(
 			// See LP1876097
 			return nil, apiservererrors.ErrPerm
 		}
-		settings, err = u.readLocalApplicationSettings(ctx, relUUID, tag, authTag.(names.UnitTag))
+		unitName, err := coreunit.NewName(authTag.Id())
```

Comment:
Maybe not something for right now, but the `ReadSettings` function above this says in the comment that using this function to read application settings is deprecated and `ReadLocalApplicationSettings` should be used instead. We could analyse the agents calls to this to see if its ever used with the application tag, if not, we could return an error here instead.